lib/DXIL/DxilMetadataHelper.cpp

```diff
@@ -1665,49 +1665,46 @@ void DxilMDHelper::LoadDxilEntryProperties(const MDOperand &MDO,
 
 MDTuple *
 DxilMDHelper::EmitDxilFunctionProps(const hlsl::DxilFunctionProps *props,
-                                   const Function *F) {
+                                    const Function *F) {
   bool bRayAttributes = false;
-  Metadata *MDVals[30];
-  std::fill(MDVals, MDVals + _countof(MDVals), nullptr);
-  unsigned valIdx = 0;
-  MDVals[valIdx++] = ValueAsMetadata::get(const_cast<Function*>(F));
-  MDVals[valIdx++] = Uint32ToConstMD(static_cast<unsigned>(props->shaderKind));
+  std::vector<Metadata *> MDVals;
```

Comment:
Should we use a SmallVector here instead to avoid growing? If we know the maximum possible, or even just a reasonable common size a SmallVector should have a significant performance benefit.